src/works/hck3791/components/hours-component.js

```diff
@@ -0,0 +1,184 @@
+import { LitElement, html } from 'lit-element';
+import { nothing } from 'lit-html';
+import { HOURS } from '../styles/styles';
+import { employees } from '../emp';
+
+class HoursComponent extends LitElement {
+
+    static get styles() {
+        return [
+            HOURS
+        ]
+    }
+
+    static get properties() {
+        return {
+            employees: { type: Array },
+            proyects: { type: Array },
+            years: { type: Array },
+            hours: { type: Array }
+        }
+    }
+
+    constructor() {
+        super();
+        this.employees = Object.keys(employees);
+        this.proyects = ['Proyect 1', 'Proyect 2'];
+        this.years = [2020, 2021, 2022];
+        this.months = ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio', 'Julio', 'Agosto', 'Septiembre', 'Octubre', 'Noviembre', 'Diciembre'];
+        this.hours = [];
+    }
+
+    generateReport() {
+        let emp = this.shadowRoot.getElementById('employees');
+        let pro = this.shadowRoot.getElementById('proyects');
+        let y = this.shadowRoot.getElementById('years');
+        let employee = emp.options[emp.selectedIndex].text;
+        let proyect = pro.options[pro.selectedIndex].text;
+        let year = y.options[y.selectedIndex].text;
+        let reset = this.shadowRoot.querySelectorAll('.data');
+
+        if(employee != '' && proyect != '' && year != '') {
+
+            console.log(true);
+            
+            let info = employees[employee][proyect][year];
+
+            if(info != null) {
+
+                console.log(info);
+                let rows = this.shadowRoot.querySelectorAll('TR');
+
+                /** Reset cells values **/
+                
+                for(let r = 0; r < reset.length; r++) {
+                    reset[r].innerText = "";
+                }
+            
+
+                for(let i = 0; i < info.length; i++) {
+                    for(let r = 0; r < rows.length; r++) {
+                        if(info[i].month == rows[r].getAttribute('id')) {
+                            let row =  rows[r];
+                            let cells = row.querySelectorAll('.data');
+                            for(let c = 0; c < cells.length; c++) {
+                                cells[c].innerText = info[i].hours[c];
+                            }                    
+                        }
+                    }
+                }
+
+            } else {
+                /** Reset cells values **/
+                
+                for(let r = 0; r < reset.length; r++) {
+                    reset[r].innerText = "";
+                }
+            }
+            
+
+        } else {
+            console.log(false);
+            /** Reset cells values **/
+            
+            for(let r = 0; r < reset.length; r++) {
+                reset[r].innerText = "";
+            }
+        }    
+        
+    }
+
+    render() {
+        return html`
+        <div class="container">
+            <main>
```

Comment:
Al ser un componente que va a ir dentro de una vista, no es necesario definir tantas etiquetas HTML5.
Habría que revisar qué etiquetas van a ir por encima pero seguramente lo correcto sería un ```<section>```

```<article>``` y ```<main>``` las suprimiría.

[HTML5 Semantic Tags: What Are They and How To Use Them!](https://www.semrush.com/blog/semantic-html5-guide/)